src/Cns.ts

```diff
@@ -57,14 +54,37 @@ export default class Cns extends NamingService {
     );
   }
 
+  static async getNetworkConfigs(config?: { url: string } | { provider: Provider } ): Promise<{ network: CnsSupportedNetworks, provider: Provider } | undefined> {
+    if (!config) {
+      return undefined;
+    }
+    try {
+      const provider = hasProvider(config) ? config.provider : new FetchProvider(NamingServiceName.CNS, config.url);
+      const networkId = await provider.request({method: "net_version"}) as number;
+      const networkName = EthereumNetworksInverted[networkId];
```

Comment:
I don't think it's necessary create an additional map to find network name. Just use EthereumNetworks.